yare-engine/src/main/java/com/sabre/oss/yare/engine/executor/DefaultRulesExecutor.java

```diff
@@ -85,11 +98,11 @@ public ExecutionContext proceed(ExecutionContext context) {
                 : configuration.isCrossProductMode() ? new CrossProductFactTupleIterator(groupedFact) : new SingleInstanceFactTupleIterator(groupedFact);
 
         if (configuration.isSequentialMode()) {
-            while (iterator.hasNext()) {
+            while (iterator.hasNext() && !stopEvaluation.get()) {
```

Comment:
I am wondering if stop processing from client should always mean: stop for all facts and all rules. Maybe it should be configurable in `stopProcessing()` method in `EngineController`?

Reply:
Postponed, but in currently implementation is not big deal to add new strategies.